packages/godwoken/index.d.ts

```diff
@@ -0,0 +1,141 @@
+import { HexNumber, Hash, Transaction, Script } from "@ckb-lumos/base";
+export interface SyncParam {
+  reverts: L1Action[];
+  updates: L1Action[];
+  next_block_context: NextBlockContext;
+}
+
+export interface L1Action {
+  transaction_info: TransactionInfo;
+  header_info: HeaderInfo;
+  context: SubmitTxs | StartChallenge | CancelChallenge | Revert;
```

Comment:
Using [discriminating unions](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions) here would be better.

Reply:
Updated.